edit-post/store/selectors.js

```diff
@@ -16,24 +21,42 @@ export function getEditorMode( state ) {
 }
 
 /**
- * Returns the current active panel for the sidebar.
+ * Returns true if the editor sidebar is opened.
  *
- * @param {Object} state Global application state.
+ * @param {Object} state Global application state
+ * @return {boolean}     Whether the editor sidebar is opened.
+ */
+export function isEditorSidebarOpened( state ) {
+	const activeGeneralSidebar = getPreference( state, 'activeGeneralSidebar', null );
+
+	return [ 'edit-post/document', 'edit-post/block' ].includes( activeGeneralSidebar );
+}
+
+/**
+ * Returns true if the plugin sidebar is opened.
  *
- * @return {string} Active sidebar panel.
+ * @param {Object} state Global application state
+ * @return {boolean}     Whether the plugin sidebar is opened.
  */
-export function getActiveEditorPanel( state ) {
-	return getPreference( state, 'activeSidebarPanel', {} ).editor;
+export function isPluginSidebarOpened( state ) {
+	const activeGeneralSidebar = getPreference( state, 'activeGeneralSidebar', null );
+
+	return Boolean( getSidebarSettings( activeGeneralSidebar ) );
 }
 
 /**
- * Returns the current active plugin for the plugin sidebar.
+ * Returns the current active general sidebar name.
+ *
+ * @param {Object} state Global application state.
  *
- * @param  {Object}  state Global application state
- * @return {string}        Active plugin sidebar plugin
+ * @return {?string} Active general sidebar name.
  */
-export function getActivePlugin( state ) {
-	return getPreference( state, 'activeSidebarPanel', {} ).plugin;
+export function getActiveGeneralSidebarName( state ) {
+	const activeGeneralSidebar = getPreference( state, 'activeGeneralSidebar', null );
+
+	return activeGeneralSidebar && ( isEditorSidebarOpened( state ) || isPluginSidebarOpened( state ) ) ?
```

Comment:
So... `activeGeneralSidebar` is not always expected to align with the value of `isEditorSidebarOpened( state ) || isPluginSidebarOpened( state )`? When does this occur? When someone tries to activate a sidebar for which there are no settings? Should that be allowed / optimized for?

Reply:
Maybe we should a commit in this PR that was not yet merged https://github.com/WordPress/gutenberg/pull/5528 to also change this includes.

Reply:
I had an early return when there is no value set. It can be now removed as I created those two helper methods.
The things get complex with the plugins when you activate a plugin sidebar, leave the page and deactivate plugin. The value is persisted so next time you visit Gutenberg it tries to load the sidebar but there is no code there. That’s way isPluginSidebarOpen does additional checks. I hope to further optimize it once the new plugin API is finalized. We could also reset the value when about to render on initial page load. If you have better ideas let us know 😃